src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java

```diff
@@ -0,0 +1,826 @@
+   * @throws JSONException if unable to get Cumulocity password from configuration
+   */
+  public String getCumulocityBootstrapPassword() throws JSONException {
+    return configurationObject
+        .getJSONObject(CONFIG_FILE_CUMULOCITY_KEY)
+        .getString(CONFIG_FILE_BOOTSTRAP_PASSWORD_KEY);
+  }
+
+  /**
+   * Get the configured Cumulocity device password.
+   *
+   * @return Cumulocity device password
+   * @throws JSONException if unable to get Cumulocity password from configuration
+   */
+  public String getCumulocityDevicePassword() throws JSONException {
+    return configurationObject
+        .getJSONObject(CONFIG_FILE_CUMULOCITY_KEY)
+        .getString(CONFIG_FILE_DEVICE_PASSWORD_KEY);
+  }
+
+  /**
+   * Set the configured Cumulocity device password
+   *
+   * @param devicePassword Cumulocity device password
+   * @throws JSONException if unable to set Cumulocity device password in configuration
+   */
+  public void setCumulocityDevicePassword(String devicePassword) throws JSONException {
+    configurationObject
+        .getJSONObject(CONFIG_FILE_CUMULOCITY_KEY)
+        .put(CONFIG_FILE_DEVICE_PASSWORD_KEY, devicePassword);
+    trySave();
+  }
+
+  /**
+   * Gets the configured Cumulocity host.
+   *
+   * @return Cumulocity host
+   * @throws JSONException if unable to get Cumulocity host from configuration
+   */
+  public String getCumulocityHost() throws JSONException {
+    return configurationObject
+        .getJSONObject(CONFIG_FILE_CUMULOCITY_KEY)
+        .getString(CONFIG_FILE_HOST_KEY);
+  }
+
+  /**
+   * Get the configured connector log level from the configuration.
+   *
+   * @return connector log level
+   * @throws JSONException if unable to get connector log level from configuration
+   */
+  public int getConnectorLogLevel() throws JSONException {
+    return configurationObject
+        .getJSONObject(CONFIG_FILE_CONNECTOR_KEY)
+        .getInt(CONFIG_FILE_LOG_LEVEL_KEY);
+  }
+
+  /**
+   * Get the configuration value for UTF-8 String support, this is required for supporting non-ASCII
+   * characters. When users leave out the JSON key, this function will return false.
+   *
+   * @return boolean for support for UTF-8 Strings
+   */
+  public boolean getStringUtf8Support() {
+    boolean isUtf8Supported = false;
+    try {
+      isUtf8Supported =
+          configurationObject
+              .getJSONObject(CONFIG_FILE_CONNECTOR_KEY)
+              .getBoolean(CONFIG_FILE_UTF8_STRING_SUPPORT_KEY);
+    } catch (JSONException e) {
+      Logger.LOG_DEBUG(CONFIG_FILE_UTF8_STRING_SUPPORT_KEY + " not found in configuration file.");
+    }
+    return isUtf8Supported;
+  }
+
+  /**
+   * Get the queue data string enabled setting from the configuration.
+   *
+   * @return queue data string enabled setting
+   * @throws JSONException if unable to get queue data string enabled setting from configuration
+   */
+  public boolean getQueueDataStringEnabled() throws JSONException {
```

Comment:
I don't think the exceptions should be handled here in the config file anymore. All fields are now present by default, so there should not really be a scenario where this issue occurs. The method already does check for the presence of the field and uses the default if the field is not there.

Reply:
The handler does not use a default value if there is an exception. The connector will fail initialization and it will show an error and shut down. This is because the user most likely has manually modified the value in the file, and it either is the incorrect type or is not formatted properly. In cases like this, I think it is imperative to inform the user because their expectation is that it will perform to their configuration, and simply defaulting the value may go missed (and a default is handled in the event that the field is just completely missing).